model/src/market/proposal.rs

```diff
@@ -35,6 +39,50 @@ pub struct Proposal {
     pub prev_proposal_id: Option<String>,
 }
 
+fn choose_payment_platform(
+    demand_platforms: Vec<String>,
+    offer_platforms: Vec<String>,
+) -> Result<String, ErrorMessage> {
+    let mut platforms = HashSet::new();
+    for platform in demand_platforms {
+        platforms.insert(platform);
+    }
```

Comment:
This could be simplified if you use `std::iter::FromIterator`
```suggestion
    let platforms  = HashSet::from_iter(demand_platforms);
```